Project overview screen

arnekuchenbecker opened the pull request.

kittyahy commented:
General things: 
Try: 
1. Rezept hinzufügen
2. Alternativrezept hinzufügen
3 .Hauptrezept löschen 
 -> keines wird mehr angezeigt, kann das Alternativrezept nicht mehr hinzufügen

- Ansicht von wann bis wann intolerante Personen da sind ist nicht da 
- Rezept und Alternativrezept tauschen geht nicht (sollte das gehen?)

kittyahy commented:
To my previous comment, I think we can make issues out of most of the things if you want

arnekuchenbecker commented:
Fixed 3. and 6., implemented 7. (but created an issue to prettify that view as the cards look kind of ugly now --> #100), created an issue for 1. (#97), 2. (#98) and 5. (#99) and did not do anything for 4. because I dont think that is very useful (if you are cooking an alternative as the main recipe why would you want the main recipe as alternative seeing as it has MORE allergens than the alternative)

kittyahy commented:
I forgot to finish my review last time, sorry

kittyahy commented:
I like the other changes

arnekuchenbecker commented on app/src/main/java/com/scouts/kitchenplaner/model/usecases/EditMealPlan.kt:
```diff
@@ -18,23 +18,39 @@ package com.scouts.kitchenplaner.model.usecases
         recipeManagementRepository.chooseMainRecipeForMealSlot(
             project.id,
-            recipe.id,
+            recipe.id ?: return,
```
RecipeStub::id is Nullable for some reason. However the repository method requires a Non-Null Long as a parameter (as it doesn't really make sense to explicitely store that no recipe has been assigned). If the passed RecipeStub has a Null id, the repository thus can't be called with any kind of useful parameter and we instead return out of the model method

arnekuchenbecker replied:
If the question was just "What does ```xyz ?: return``` do?", then the short answer is that if xyz is not null, it is passed as the parameter, otherwise we return out of the method (in this case ```selectMainRecipeForMealSlot()```)

arnekuchenbecker commented on app/src/main/java/com/scouts/kitchenplaner/ui/navigation/NavHostProjects.kt:
```diff
@@ -33,24 +34,19 @@ private const val RECIPE_TO_COOK = "recipeToCook"
     modifier: Modifier = Modifier,
     projectNavController: NavHostController,
-    id: Long = 0L,
+    project: Project,
```
Yes, however we kind of have a weird mishmash of Navigation and Composables here. The outer composable (ProjectLayout) already needs the project (for the title and image, but also for the settings sidebar) and it doesn't really make sense to have two flows of the same project coming out of the database (as this would lead to duplicate updates of the composable hierarchy). Then we navigate further into one of the screens. To avoid above mentioned issues, we pass the project down through the navigation here. This is maybe something that can change once we change navigation for the OnRecipeCreated callback, but for now this should (I think) stay this way

arnekuchenbecker commented on app/src/main/java/com/scouts/kitchenplaner/ui/view/allergendialog/AllergenCard.kt:
```diff
@@ -38,13 +38,21 @@ import com.scouts.kitchenplaner.ui.view.DeleteButton
+    onItemDelete: (Pair<String, Boolean>) -> Unit,
+    toBeDeleted: Boolean,
+    toggleExpand: () -> Unit,
+    expand: Boolean
```
Added this, however now the allergen cards are looking terrible. Created issue #109 to fix this
